usaspending_api/recipient/delta_models/recipient_profile.py

```diff
@@ -1,26 +1,433 @@
+            ELSE CONCAT('name-', COALESCE(fpds.awardee_or_recipient_legal, fabs.awardee_or_recipient_legal, '')) END
+        )), '^(\.{{8}})(\.{{4}})(\.{{4}})(\.{{4}})(\.{{12}})$', '\$1-\$2-\$3-\$4-\$5') AS recipient_hash,
+        COALESCE(fpds.awardee_or_recipient_uniqu, fabs.awardee_or_recipient_uniqu) AS recipient_unique_id,
+        COALESCE(fpds.ultimate_parent_unique_ide, fabs.ultimate_parent_unique_ide) AS parent_recipient_unique_id,
+        COALESCE(fpds.awardee_or_recipient_uei, fabs.uei) AS uei,
+        COALESCE(fpds.ultimate_parent_uei, fabs.ultimate_parent_uei) AS parent_uei,
+        CASE
+            WHEN tn.type IN ('A', 'B', 'C', 'D')      THEN 'contract'
+            WHEN tn.type IN ('02', '03', '04', '05')  THEN 'grant'
+            WHEN tn.type IN ('06', '10')              THEN 'direct payment'
+            WHEN tn.type IN ('07', '08')              THEN 'loans'
+            WHEN tn.type IN ('09', '11')              THEN 'other'     -- collapsing insurance into other
+            WHEN tn.type LIKE 'IDV%'                  THEN 'contract'  -- collapsing idv into contract
+            ELSE NULL
+        END AS award_category,
+        CASE
+            WHEN COALESCE(fpds.ultimate_parent_uei, fabs.ultimate_parent_uei) IS NOT NULL THEN 'C'
+            ELSE 'R' END AS recipient_level,
+        tn.action_date,
+        CAST(COALESCE(CASE
+            WHEN tn.type IN('07','08') THEN tn.original_loan_subsidy_cost
+            ELSE tn.federal_action_obligation
+        END, 0) AS NUMERIC(23, 2)) AS generated_pragmatic_obligation
+    FROM
+        raw.transaction_normalized as tn
+    LEFT OUTER JOIN raw.transaction_fpds as fpds ON tn.id = fpds.transaction_id
+    LEFT OUTER JOIN raw.transaction_fabs as fabs ON tn.id = fabs.transaction_id
+    WHERE
+        tn.action_date >= '2007-10-01'
+        AND tn.type IS NOT NULL
+    );""",
+    #
+    # --------------------------------------------------------------------------------
+    # -- Step 2, Create the new table and populate with 100% of combinations
+    # --------------------------------------------------------------------------------
+    f"""CREATE OR REPLACE TABLE temporary_restock_recipient_profile (
+        id BIGINT,
+        recipient_level character(1) NOT NULL,
+        recipient_hash STRING,
+        recipient_unique_id STRING,
+        uei STRING,
+        parent_uei STRING,
+        recipient_name STRING,
+        unused BOOLEAN,
+        recipient_affiliations ARRAY<STRING>,
+        award_types ARRAY<STRING>,
+        last_12_months NUMERIC(23,2),
+        last_12_contracts NUMERIC(23,2),
+        last_12_grants NUMERIC(23,2),
+        last_12_direct_payments NUMERIC(23,2),
+        last_12_loans NUMERIC(23,2),
+        last_12_other NUMERIC(23,2),
+        last_12_months_count INT
+    ) USING DELTA;
+    """,
+    """INSERT INTO temporary_restock_recipient_profile (
+    SELECT
+        monotonically_increasing_id() AS id,
```

Comment:
My understanding is that `monotonically_increasing_id()` is guaranteed unique for a single query / table (assuming it falls within the 8 billion constraint for partitions), but that the value is not guaranteed unique between tables / queries.

Since you have `monotonically_increasing_id() AS id` that means you could have multiple of the same `id` field in this query with UNION statements. 

For example, 
```
SELECT monotonically_increasing_id()
UNION ALL 
SELECT monotonically_increasing_id();
```
yields 
```
0
0
```

Reply:
There isn't a way to duplicate the auto-incrementing ids from postgres, so for the time being I am just mocking the IDs in delta, and when we move the data back into postgres we should be able to not move the id field and let the postgres table automatically fill in the id field. We don't use the id field for anything as far as i'm aware, so it should be fine for it to be dummy data.